### DISCUSSION TITLE
Cannot Find LaTeX Error

### DISCUSSION THREAD

**Discussion by anaclumos** [facebook/docusaurus#10275] [General]
When I compile [my website](https://github.com/anaclumos/extracranial), with `pnpm qa:build`, it gives me:

```bash
"commentAtEnd": Use of % comment without a terminating newline. LaTeX would thereby comment out the end of math mode (e.g. $), causing an error.
```

But I cannot figure out which file has the error. Any suggestions?

**Comment by slorber** [comment]
What does it mean "it gives me"

If it's a log, please post the full console output

**Reply by anaclumos** [reply]
https://asciinema.org/a/5hFf804KOLGrYJwfR6MqznScr

Here is the full console output. Please check from 1:10-ish.

**Reply by slorber** [reply]
So what I understand is that the full log looks like this:

```
● Client                           building (11%) 0/1 entries 8436/8466 dependencies 96/3319 modules 1143 active                                     
                                                                                                                                                     
● Client                           building (11%) 0/1 entries 8454/8484 dependencies 96/3319 modules 1151 active                                     
 mdx-loader › docs/pages/0206 Reverse Linked List.md                                                                                                 
                                                                                                                                                     
● Server                           building (11%) 0/1 entries 5849/5873 dependencies 107/3325 modules 1138 active                                    
 mdx-loader › docs/pages/TossFace Playground.md                                                                                                      
                                                                                                                                                     
LaTeX-incompatible input and strict mode is set to 'warn': % comment has no terminating newline; LaTeX would fail because of commenting the end of ma
th mode (e.g. $) [commentAtEnd]                                                                                                                      
LaTeX-incompatible input and strict mode is set to 'warn': % comment has no terminating newline; LaTeX would fail because of commenting the end of ma
th mode (e.g. $) [commentAtEnd]                                                                                                                      
⌛ Done in 00h 01m 17s.                                                                                                                               
-----------------------------------------------------------------------------------------------------------------------------------------------------
~/Downloads/www (main*) » 
```

I have no idea what is going on but I think I already saw this on a localized Docusaurus build in Chinese/Korean. Maybe there's a syntax problem on your site, but we can't troubleshoot that without a minimal repro containing the problematic source. Does it prevent the site from building?

**Reply by anaclumos** [reply]
I see. The website still succeeds to build but I just wanted to report and look for the error source. Is there a way to make the error more descriptive so that I can pinpoint the erroneous document? I currently have 5000+ documents 😭

**Reply by slorber** [reply]
Unfortunately, if LaTeX only warns and doesn't fail fast, I don't have any solution apart the dicothomy (removing half of your docs at each step) or putting logger/debugging breakpoints on LaTeX code that produces the warning.

**Reply by Josh-Cena** [reply]
FYI, you have a lot of unescaped `$` that are turning into math formulae, and I suspect the problem is in one of them. For example, this page: https://cho.sh/r/2D85E7

I searched this: `repo:anaclumos/extracranial /%.*\$/ (path:**/*.md OR path:**/*.mdx)` which has five hits. Perhaps it's in one of those? You can first try to do a local search for files that contain both `%` and `$`.

**Reply by anaclumos** [reply]
No luck :/

I'll just keep it like this lol

Will revisit when this becomes a problem again

**Comment by OpportunityLiu** [comment]
Currently, my solution is adding breakpoint at `katex/dist/katex.js:448`, and run `docusaurus start` with debugger. Then you can find the source from call stack.

![image](https://github.com/user-attachments/assets/2525902f-56f0-4d8c-968a-50b211d3ba42)
